### PR TITLE
fix: Change the redis client library from redigo to go-redis

### PR DESCRIPTION
**Pull request by uzziahlin**
Change the redis client library from redigo to go-redis

### CODE REVIEW

File: redis/redis.go [hunk 1]
```diff
@@ -17,20 +17,21 @@ package redis
...
-	berror "github.com/beego/beego-error/v2"
-	"github.com/gomodule/redigo/redis"
 )
 
-// DefaultKey defines the collection name of redis for the cache adapter.
-var DefaultKey = "beecacheRedis"
+var (
+	// DefaultKey defines the collection name of redis for the cache adapter.
+	DefaultKey = "beecacheRedis"
+	ScanCount  = int64(1 << 10) // aka 1024
```
Comment (flycash) [inline redis/redis.go]: remove this one and then add a new field `scanCount` to Cache structure, and the default value of scanCount is 1024

File: redis/redis.go [hunk 2]
```diff
@@ -17,20 +17,21 @@ package redis
...
+)
 
 // Cache is Redis cache adapter.
 type Cache struct {
-	p        *redis.Pool // redis connection pool
+	client   redis.Cmdable // redis client
```
Comment (flycash) [inline redis/redis.go]: do we still need the conninfo? can we remove it?

**Comment by flycash** [comment]
rename the `mock/redis_mock.go` to `mock/redis_mock_gen_test.go` so that users can not access MockXXXX

**Comment by flycash** [comment]
You can run the `make fmt` before pushing the changes to remote repository. Don't forget to add this merge request into .CHANGELOG file.

Please check the Deepsource and fix the new issues introduced by this PR.

File: redis/redis.go (hunk 2 above)
Comment (uzziahlin) [reply redis/redis.go]: I don't think it's needed either, so let me delete it

File: redis/redis.go (hunk 1 above)
Comment (uzziahlin) [reply redis/redis.go]: This is a good idea.

**Comment by uzziahlin** [comment]
> rename the `mock/redis_mock.go` to `mock/redis_mock_gen_test.go` so that users can not access MockXXXX将 `mock/redis_mock.go` 重命名为 `mock/redis_mock_gen_test.go` ，以便用户无法访问 MockXXXX

If I rename this file to mock/redis_mock_gen_test.go, I can't seem to introduce it in redis_test.go

**Comment by flycash** [comment]
> > rename the `mock/redis_mock.go` to `mock/redis_mock_gen_test.go` so that users can not access MockXXXX将 `mock/redis_mock.go` 重命名为 `mock/redis_mock_gen_test.go` ，以便用户无法访问 MockXXXX
> 
> If I rename this file to mock/redis_mock_gen_test.go, I can't seem to introduce it in redis_test.go

ohhh, sorry, I made a mistake. You can put it into cache/internal/mock/redis_mock.go

**Comment by flycash** [comment]
I will merge it and I fix some tiny issues by myself, thank you

**Pull request merged**